soccer/gameplay/tactics/coordinated_pass.py

```diff
@@ -175,15 +189,37 @@ def use_chipper(self, value):
     def on_exit_running(self):
         self.remove_subbehavior('receiver')
 
-    def on_enter_kicking(self):
+    # Funciton to find if ball is moving towards receive point
+    def ball_heading_to_receive_point(self):
+        ball_velocity = main.ball().vel;
+        if ball_velocity.mag() > 0.1:
+            ball_velocity_unit = [ball_velocity.x/ball_velocity.mag(), ball_velocity.y/ball_velocity.mag()]
+            path_vector = main.ball().pos - self.receive_point
+            path_unit = [path_vector.x/path_vector.mag(), path_vector.y/path_vector.mag()]
+            cross_product_mag = abs(ball_velocity_unit[0] * path_unit[1] -
+                ball_velocity_unit[1] * path_unit[0])
+            distance = path_vector.mag()
+            if distance < 0.3:
+                return True # If the ball is already close to the receive point but not heading towards it, dont fail
+            if (distance > 0.5 and cross_product_mag > 0.1) or (distance < 0.5 and cross_product_mag > 0.4) or ball_velocity.mag() < 0.05:
+                return False # If ball is too off course or too slow fail the pass
+        return True
+
+    # Funciton that decides if an opponent robot is too close to the kicking robot
+    def opponent_too_close(self):
```

Comment:
I'm kind of sketched out by the fact that this is a thing. Seems pretty questionable, but I'm alright with it as a hotfix.

Maybe put a TODO so we know to look there later.

Reply:
I disagree with stopping a pass because an opponent is too close. If we consider the coordinated pass to fail, we would transition to the next play / situation, which would probably include a kick anyway